providers/base/units/audio/jobs.pxu

```diff
@@ -695,32 +695,30 @@ _summary: Check that at least one audio playback device exits
 plugin: shell
 category_id: com.canonical.plainbox::audio
 flags: also-after-suspend
-imports: from com.canonical.plainbox import manifest
 requires:
-  manifest.has_audio_playback == 'True'
+  audio_card.Playback == 'supported'
```

Comment:
I may be very wrong here, but doesn't the resource job output `playback : 1`? Genuine question, how does `Playback` arrive here?

Reply:
Checkbox could create an entry for a resource job, so other jobs can call variables from the entry:
1. Define an ID in resource.pxu.
```
id: audio_card
estimated_duration: 0.05
plugin: resource
_summary: Collect information about the audio card
_description: Gets audio resource info from /proc/asound/pcm
command: audio_card_resource.py
```

2. Create the entry in test-plan.pxu (so before the jobs in the test-plan start, audio_card will run first and gather audio card info).
```
bootstrap_include:
 audio_card
```
Then, in job.pxu, we can use `requires` to set preconditions. Which variable should be set to what value to have a go.
```
requires:
  audio_card.playback == 'supported'
```

Reply:
the thing that puzzles me a bit is that I always thought that it is the printer that defines what is inside the resources, and the printer that you have created prints either 1 or 0, not supported and unsupported. Also I thought that resource entities had to be newline separated. I am not up to the task of saying if this is correct or not though. @pieqq may know more here

Reply:
I agree with @Hook25, there is a problem here.

If I run the `audio/detect_sources` job, for instance, which has been modified to use the `audio_card` resource requirements, I get this:

```
$ checkbox-cli run com.canonical.certification::audio/detect_sources
(...)
==============[ Running job 3 / 3. Estimated time left: 0:00:01 ]===============
----------------------------[ audio/detect_sources ]----------------------------
ID: com.canonical.certification::audio/detect_sources
Category: com.canonical.plainbox::audio
Job cannot be started because:
 - resource expression "audio_card.capture == 'supported'" evaluates to false
Outcome: job cannot be started
```

And this is expected, because if I run the `audio_card` resource job itself, I get that:

```
$ checkbox-cli run com.canonical.certification::audio_card
------------------[ Collect information about the audio card ]------------------
ID: com.canonical.certification::audio_card
Category: com.canonical.plainbox::uncategorised
... 8< -------------------------------------------------------------------------
Card: 00
Device: 00
Name: ALC887-VD Analog
Playback: 1
Capture: 1

(...)

Card: 03
Device: 00
Name: USB Audio
Capture: 1

------------------------------------------------------------------------- >8 ---
Outcome: job passed

```

Two issues here:

1. In the `audio_card` resource job, there is no `capture` field (but there is a `Capture` field!)
2. In the `audio_card` resource job, the `Capture` field is set to `1` when capture is supported, not to `supported`

I have a few recommendations:

- If you look at the scripts used in other resource jobs (for instance `cpuinfo_resource.py`), you will see that the fields they provide are all lowercases, and the values for (un)supported status is usually `supported`/`unsupported`:

```
$ ./cpuinfo_resource.py 
bogomips: 6385
cache: 6291456
count: 4
model: Intel(R) Core(TM) i5-4570 CPU @ 3.20GHz
model_number: 6
model_revision: 3
model_version: 60
other: fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush dts acpi mmx fxsr sse sse2 ss ht tm pbe syscall nx pdpe1gb rdtscp lm constant_tsc arch_perfmon pebs bts rep_good nopl xtopology nonstop_tsc cpuid aperfmperf pni pclmulqdq dtes64 monitor ds_cpl vmx smx est tm2 ssse3 sdbg fma cx16 xtpr pdcm pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand lahf_lm abm cpuid_fault epb invpcid_single pti ssbd ibrs ibpb stibp tpr_shadow vnmi flexpriority ept vpid ept_ad fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid xsaveopt dtherm ida arat pln pts md_clear flush_l1d
platform: x86_64
speed: 3600
type: GenuineIntel
governors: conservative ondemand userspace powersave performance schedutil
scaling: supported
```

I would recommend you use the same convention for clarity.

(I'm gonna add a few other comments in a separate review)